src/include/cursor.h

```diff
@@ -527,6 +527,36 @@ struct __wt_cursor_table {
     WT_CURSOR **idx_cursors;
 };
 
+struct __wt_cursor_version {
+    WT_CURSOR iface;
+
+    WT_CURSOR *hs_cursor;    /* Queries of history cursor. */
+    WT_CURSOR *table_cursor; /* Queries of regular table cursor. */
```

Comment:
With respect to the previous comment about `table` vs. `file` -- is this always a table cursor, or do you mean "object" here?